src/parser.cpp

```diff
@@ -56,9 +56,7 @@ namespace ada::parser {
 
     // Optimization opportunity. Most websites do not have fragment.
     std::optional<std::string_view> fragment = helpers::prune_fragment(url_data);
-    if (fragment.has_value()) {
-      url.update_unencoded_base_hash(*fragment);
-    }
+    // We add it last.
```

Comment:
Can you add the reasoning to this line here? I assume it is faster due to the unnecessary if statement checks when parsing it, but it might be good to document it in here.

Reply:
Comment added.